lib/clouds/shared_functions.py

```diff
@@ -813,13 +822,342 @@ def populate_interface(self, obj_attr_list) :
                     except Exception, msg :
                         _fmsg = "Could not lookup interface " + iface + " for hostname " + hostname + " (probably bad /etc/hosts): " + str(msg)
                         raise CldOpsException(_fmsg, 1295)
+                    
+    @trace
+    def is_cloud_image_uuid(self, imageid) :
+        '''
+        TBD
+        '''
+        if imageid == "to_replace" :
+            return False
+        
+        if self.get_description() == "Amazon Elastic Compute Cloud" :
+            if len(imageid) > 4 :
+                if imageid[0:4] == "ami-" :
+                    if is_number(imageid[5:], True) :
+                        return True
+
+        if self.get_description() == "Cloudbench SimCloud" or self.get_description() == "Cloudbench NoOpCloud" :
+            if len(imageid) == 36 and imageid.count('-') == 4 :
+                return True
+        
+        if self.get_description() == "OpenStack Cloud" :
+            if len(imageid) == 36 and imageid.count('-') == 4 :
+                return True
+
+        if self.get_description() == "SoftLayer Cloud" :
+            if len(imageid) == 7 and is_number(imageid) :
+                return True
+
+        if self.get_description() == "Google Compute Engine" :
+            if len(imageid) == 19 and is_number(imageid) :
+                return True
+
+        if self.get_description() == "DigitalOcean Cloud" :
```

Comment:
This is a pretty crazy set of conditionals. Can't this be in a configuration file instead of being hard-code like this? If we implement a new cloud adapter it means I would have to go into this file and manually perform steps here that are not contained inside the individual cloud adapter.